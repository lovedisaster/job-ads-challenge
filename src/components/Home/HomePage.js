import React, { useCallback, useEffect } from "react";
import PropTypes from "prop-types";
import { AdCard, Heading } from "./Home.style";
import { Button } from "../shared/atoms/buttons/Buttons";
import ActionTypes from "../../actions/ActionTypes";

const HomePage = (props) => {

  const buyNowOnclick = useCallback((stepData) => {
    props.dispatch({
      type: ActionTypes.SAVE_STEP_DATA,
      payload: stepData,
    });
    props.dispatch({
      type: ActionTypes.ADD_TO_CART,
      payload: stepData,
    });
    props.dispatch({ type: ActionTypes.NEXT_STEP });
  });

  useEffect(() => {
    props.dispatch({type: ActionTypes.PAGE_LOADED});
  },[]);

  return (
    <div className="container">
      <br />
      <br />
      <br />
      <br />
      <br />
      <Heading className="text-center">Job Ads</Heading>
      <br />
      <br />
      <br />
      <div className="row">
        {props.state.initState.ads.map((item) => {
          return (
            <div key={item.code} className="col-4 text-center">
              <AdCard>
                <h4 className="ad-name">{item.name}</h4>
                <h2 className="price">${item.price}</h2>
                <p className="desc">{item.description}</p>
                <Button
                  color={GS.colors.primaryBlue}
                  value="Buy now"
                  onClickHandler={() => buyNowOnclick(item)}
                />
              </AdCard>
            </div>
          );
        })}
      </div>
    </div>
  );
};

HomePage.propTypes = {
  context: PropTypes.object,
};

export default HomePage;
