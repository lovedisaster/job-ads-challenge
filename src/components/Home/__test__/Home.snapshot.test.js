import React from 'react';
import HomePage from '../HomePage';
import TestRenderer from 'react-test-renderer';

describe("Test ad product page", () => {
    it("Home page snapshot" , () => {
        const MockStateProps = {
               initState:{
                  ads:[
                     {
                        code:"CLASSIC",
                        name:"Classic Ad",
                        description:"Offers the most basic level of advertisement",
                        price:269.99
                     },
                     {
                        code:"STANDOUT",
                        name:"Stand out Ad",
                        description:"Allows advertisers to use a company logo and use a longer presentation text",
                        price:322.99
                     },
                     {
                        code:"PREMIUM",
                        name:"Premium Ad",
                        description:"Same benefits as Standout Ad, but also puts the advertisement at the top of the results, allowing higher visibility",
                        price:394.99
                     }
                  ]
               }
         }
        const RenderedHomePage = TestRenderer.create(<HomePage state={MockStateProps} dispatch={() => {}}></HomePage>).toJSON();
        expect(RenderedHomePage).toMatchSnapshot();
    });
});