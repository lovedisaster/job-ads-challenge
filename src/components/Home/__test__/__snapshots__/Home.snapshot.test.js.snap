// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ad product page Home page snapshot 1`] = `
<div
  className="container"
>
  <br />
  <br />
  <br />
  <br />
  <br />
  <h1
    className="Homestyle__Heading-bedjur-0 kQKohX text-center"
  >
    Job Ads
  </h1>
  <br />
  <br />
  <br />
  <div
    className="row"
  >
    <div
      className="col-4 text-center"
    >
      <div
        className="Homestyle__AdCard-bedjur-1 cZDLhm"
      >
        <h4
          className="ad-name"
        >
          Classic Ad
        </h4>
        <h2
          className="price"
        >
          $
          269.99
        </h2>
        <p
          className="desc"
        >
          Offers the most basic level of advertisement
        </p>
        <button
          className="buttonstyles__SolidButton-e7xntx-0 gVEEdL"
          onClick={[Function]}
        >
          Buy now
        </button>
      </div>
    </div>
    <div
      className="col-4 text-center"
    >
      <div
        className="Homestyle__AdCard-bedjur-1 cZDLhm"
      >
        <h4
          className="ad-name"
        >
          Stand out Ad
        </h4>
        <h2
          className="price"
        >
          $
          322.99
        </h2>
        <p
          className="desc"
        >
          Allows advertisers to use a company logo and use a longer presentation text
        </p>
        <button
          className="buttonstyles__SolidButton-e7xntx-0 gVEEdL"
          onClick={[Function]}
        >
          Buy now
        </button>
      </div>
    </div>
    <div
      className="col-4 text-center"
    >
      <div
        className="Homestyle__AdCard-bedjur-1 cZDLhm"
      >
        <h4
          className="ad-name"
        >
          Premium Ad
        </h4>
        <h2
          className="price"
        >
          $
          394.99
        </h2>
        <p
          className="desc"
        >
          Same benefits as Standout Ad, but also puts the advertisement at the top of the results, allowing higher visibility
        </p>
        <button
          className="buttonstyles__SolidButton-e7xntx-0 gVEEdL"
          onClick={[Function]}
        >
          Buy now
        </button>
      </div>
    </div>
  </div>
</div>
`;
